packages/SwingSet/test/abandon-export.test.js

```diff
@@ -56,19 +62,29 @@ const makeTestVat = async (t, kernel, name) => {
   const kref = kernel.getRootObject(vatID);
   kernel.pinObject(kref);
   const flushDeliveries = async () => {
-    // Make a dummy delivery so the kernel will call processRefcounts().
-    // This isn't normally needed, but we're directly making syscalls
-    // outside of a delivery.
-    // If that ever stops working, we can update `makeTestVat` to return
-    // functions supporting more true-to-production vat behavior like
-    // ```js
-    // enqueueSyscall('send', target, kser([...args]), resultVPID);
-    // enqueueSyscall('subscribe', resultVPID);
-    // flushSyscalls(); // issues queued syscalls inside a dummy delivery
-    // ```
-    kernel.queueToKref(kref, 'flush', []);
+    // This test uses a really lazy+sketchy approach to triggering
+    // syscalls: it just grabs the vat's `syscall` object and invokes
+    // it, from outside a crank. We're exercising refcount-changing
+    // syscalls, but the kernel only runs processRefcounts() after a
+    // real crank (i.e. inside processDeliveryMessage and
+    // processAcceptanceMessage). We need to trigger this, but we want
+    // to avoid making a real delivery to one of our vats, so their
+    // delivery logs won't be cluttered with our dummy
+    // processRefcounts-triggering activity
+    //
+    // The previous hack was to do a queueToKref(), and then try to
+    // remove the resulting delivery from the stream.
+    //
+    // The new hack is to inject a `{ type: 'negated-gc-action' }`
+    // onto the run-queue, which is handled by processDeliveryMessage
+    // but doesn't actually deliver anything.
+    //
+    // A safer approach would be to define the vat's dispatch() to
+    // listen for some messages that trigger each of the syscalls we
+    // want to invoke
+    //
+    kernelKeeper.addToRunQueue({ type: 'negated-gc-action' });
     await kernel.run();
-    assertFirstLogEntry(t, log, 'deliver', { method: 'flush' });
   };
   return { vatID, kref, dispatch, log, syscall, flushDeliveries };
```

Comment:
I wonder if we should drop the [unused] `dispatch` result.
```suggestion
  return { vatID, kref, log, syscall, flushDeliveries };
```

Reply:
yeah, ok, I think I copied this from a test that did more with it